src/goto-instrument/cover_basic_blocks.cpp

```diff
@@ -63,6 +63,7 @@ cover_basic_blockst::cover_basic_blockst(const goto_programt &_goto_program)
 
     // set representative program location to instrument
     if(
+      !it->function.empty() &&
```

Comment:
This should be an invariant, I'd say. So I'd turn this on its head: fix the code that generates new instructions without setting their `function` field.

Reply:
Correct. Let me check how big this piece of work will be. I will then put an INVARIANT in goto_programt::update.

Reply:
+1 -- I'd always thought this was a DATA_INVARIANT().  It would certainly be Very Useful if it was.

Reply:
See https://github.com/diffblue/cbmc/pull/1843 regarding my investigations on this topic.